### PR TITLE
Log ccd errors

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1210

### Change description ###
Logging CCD error responses.

There will be a PR to log error responses from PayHub.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 1]
```diff
@@ -69,58 +77,79 @@ public void completeAwaitingDcnProcessing(
...
+                caseRef,
+                jurisdiction,
+                caseTypeId
+            );
+
+            return response;
+        } catch (FeignException ex) {
+            debugCcdException(ex, "Failed to call 'startCompleteAwaitingDcnProcessing'");
+            throw new CcdCallException(
+                format("Internal Error: start event call failed case: %s Error: %s", caseRef, ex.status()), ex
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java:109]: Shall we make it more acceptable for case worker? e.g. `Failed to start event in CCD for completing DCN processing`. Well, smth

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 2]
```diff
@@ -69,58 +77,79 @@ public void completeAwaitingDcnProcessing(
...
+                "Submitted event {} for case {}. Jurisdiction: {}. Case type ID: {}",
+                caseDataContent.getEvent().getId(),
+                caseRef,
+                jurisdiction,
+                caseTypeId
+            );
+        } catch (FeignException ex) {
+            debugCcdException(ex, "Failed to call 'submitCompleteAwaitingDcnProcessing'");
+            throw new CcdCallException(
+                format("Internal Error: submit event call failed case: %s Error: %s", caseRef, ex.status()), ex
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java:143]: same here if previous applies

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java [hunk 3]
```diff
@@ -143,10 +153,15 @@ void should_fail_when_when_submit_event_fails() {
...
-        assertThatThrownBy(() ->
-            ccdClient.completeAwaitingDcnProcessing("1231244243242343", "bulkscan", "BULKSCAN")
-        ).isSameAs(FEIGN_EXCEPTION);
+        // when
+        CcdCallException exception = catchThrowableOfType(
+            () -> ccdClient.completeAwaitingDcnProcessing("1231244243242343", "bulkscan", "BULKSCAN"),
+            CcdCallException.class
+        );
+
+        assertThat(exception.getMessage())
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java:162]: if `when` then `then` ;)

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java (hunk 1 above)
Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java:109]: Good point

**Comment by apereverzin** [comment]
Not being picked up by Jenkins

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java [hunk 4]
```diff
@@ -69,58 +77,89 @@ public void completeAwaitingDcnProcessing(
...
+                caseTypeId
+            );
+
+            return response;
+        } catch (FeignException ex) {
+            debugCcdException(ex, "Failed to call 'startCompleteAwaitingDcnProcessing'");
+            throw new CcdCallException(
+                format(
+                    "Internal Error: Failed starting event in CCD for completing payment DCN processing "
+                        + "case: %s Error: %s", caseRef,
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java:111]: Perhaps I should've mentioned it: not sure `Internal Error:` serves any purpose here. Same with another one below

**Review by doncem — APPROVED** [review]

**Review by mdayican — APPROVED** [review]

**Pull request merged**